src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1603,6 +1603,84 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   </Target>
 
+  <!--
+    ====================================================================================
+                                        _GetProjectReferencePlatformProperties
+
+    If a project is opted in via $(EnableDynamicPlatformResolution), this target builds the GetPlatforms
+    target of all existing references to get a list of all possible platforms of the referenced projects. 
+    Calls the GetCompatiblePlatform task to determine the most compatible platform for each project.
+    This prevents overbuilding a project when 'AnyCPU' is available, for example.
+
+    ======================================================================================
+  -->
+
+  <!-- Managed projects need to have PlatformTarget set for SetPlatform negotiation. Default to $(Platform), which is AnyCPU by default. -->
+  <PropertyGroup>
+    <PlatformTarget Condition="'$(EnableDynamicPlatformResolution)' == 'true' and '$(PlatformTarget)' == ''
+                                and '$(MSBuildProjectExtension)' != '.vcxproj' and '$(MSBuildProjectExtension)' != '.nativeproj'">$(Platform)</PlatformTarget>
+  </PropertyGroup>
+
+  <Target Name="_GetProjectReferencePlatformProperties"
+          Condition="'$(EnableDynamicPlatformResolution)' == 'true'
+                     and '$(BuildingInsideVisualStudio)' != 'true'
+                     and '@(_MSBuildProjectReferenceExistent)' != ''">
+
+    <!-- Allow preset SetPlatform to override this operation -->
+    <ItemGroup>
+      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SetPlatform)' != ''">
+        <SkipGetPlatformProperties>true</SkipGetPlatformProperties>
+      </_MSBuildProjectReferenceExistent>
+    </ItemGroup>
+
+    <ItemGroup>
+      <_ProjectReferencePlatformPossibilities Include="@(_MSBuildProjectReferenceExistent)" 
+                                              Condition="'%(_MSBuildProjectReferenceExistent.SkipGetPlatformProperties)' != 'true'"/>
+    </ItemGroup>
+
+    <!-- Managed Platform "source of truth" is $(PlatformTarget). For cpp it's $(Platform) -->
+    <PropertyGroup>
+      <ParentPlatform>$(PlatformTarget)</ParentPlatform>
+      <ParentPlatform Condition="'$(MSBuildProjectExtension)' == '.vcxproj' or '$(MSBuildProjectExtension)' == '.nativeproj'">$(Platform)</ParentPlatform>
+    </PropertyGroup>
+
+    <!-- If we're looking at a c++ project from a managed project, map managed platforms to native platforms. -->
+    <ItemGroup>
+      <_ProjectReferencePlatformPossibilities Condition="'$(MSBuildProjectExtension)' != '.vcxproj' and '$(MSBuildProjectExtension)' != '.nativeproj' and '%(_ProjectReferencePlatformPossibilities.IsVcxOrNativeProj)' == 'true'">
```

Comment:
So there's already default case for a cpp project receiving `AnyCPU` as the `Platform`? Where is that handled?

Reply:
Thinking on this: We shouldn't pass `Platform` to `ProjectReference` items if we don't know for sure that it can build with that (child project listing it in `Platforms` or a `PlatformTranslationTable` being defined).

If we pass it through then unmanaged platforms would be built with `Platform=AnyCPU` which both doesn't make sense and would give us an evaluation that we then need to dig into to see what AnyCPU actually converted into.

Reply:
I'm confused here. Why do you need to check that it is neither a vcxproj nor a nativeproj, but something else IsVcxOrNativeProj?

Reply:
`$(MSBuildProjectExtension)` is the current project's file extension, `'%(_ProjectReferencePlatformPossibilities.IsVcxOrNativeProj)'` is any given project reference that happens to be a vcx or nativeproj.

In english, this item does the following: "When this is a managed project, assign a default `PlatformLookupTable` metadata to all `ProjectReference` items that are `.vcxproj` or `.nativeproj`." I should probably update the comment with this clearer explanation.

For the purposes of this opt-in feature, we're assuming anything that isn't vcxproj or nativeproj is managed.

Reply:
Instead of check for vcxproj and nativeproj. You should check for supported extensions like vbproj, csproj etc.  You can have control of this list.